cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java

```diff
@@ -144,12 +145,18 @@ public void execute() {
                 Path parentPath = sourcePath.getParent();
                 if (Files.isRegularFile(resolvedFullPath) && sourcePath.toString().endsWith(BLANG_SRC_FILE_SUFFIX) &&
                         parentPath != null) {
-                    Path fileName = parentPath.getFileName();
-                    String srcPkgName = fileName != null ? fileName.toString() : "";
                     throw LauncherUtils.createLauncherException("you are trying to build a ballerina file inside a " +
                                                                         "package within a project. Try running " +
                                                                         "'ballerina build <package-name>'");
                 }
+                // Checks if the source path is a package
```

Comment:
This check shouldn't be done here in this class. This is a low-level check that should be done somewhere else. 

We are calling the "ProjectDirs.containsSourceFiles" method here. It is a class that should only be known by the compiler not the build command.

Reply:
Moved the check to the compiler with ada9c2d